plugins/nf-tower/src/main/io/seqera/tower/plugin/cli/AuthCommandImpl.groovy

```diff
@@ -0,0 +1,1354 @@
+        final line = console != null
+            ? console.readLine()
+            : new BufferedReader(new InputStreamReader(System.in)).readLine()
+        return line?.trim()
+    }
+
+    protected Map getWorkspaceDetailsFromApi(String accessToken, String endpoint, String workspaceId) {
+        try {
+            final userInfo = callUserInfoApi(accessToken, endpoint)
+            final userId = userInfo.id as String
+
+            final client = createHttpClient(accessToken)
+            final request = HttpRequest.newBuilder()
+                .uri(URI.create("${endpoint}/user/${userId}/workspaces"))
+                .GET()
+                .build()
+
+            final response = client.send(request, HttpResponse.BodyHandlers.ofString())
+
+            if (response.statusCode() != 200) {
+                return null
+            }
+
+            final json = new JsonSlurper().parseText(response.body()) as Map
+            final orgsAndWorkspaces = json.orgsAndWorkspaces as List
+
+            final workspace = orgsAndWorkspaces.find { ((Map)it).workspaceId?.toString() == workspaceId }
+            if (workspace) {
+                final ws = workspace as Map
+                return [
+                    orgName: ws.orgName,
+                    workspaceName: ws.workspaceName,
+                    workspaceFullName: ws.workspaceFullName
+                ]
+            }
+
+            return null
+        } catch (Exception e) {
+            return null
+        }
+    }
+
+    private Map getCloudEndpointInfo(String apiUrl) {
+        for (env in SEQERA_API_TO_AUTH0) {
+            final standardUrl = env.key as String
+            final legacyUrl = standardUrl.replace('://api.', '://') + '/api'
+            if (apiUrl == standardUrl || apiUrl == legacyUrl) {
+                return [isCloud: true, endpoint: env.key, auth: env.value]
+            }
+        }
+        return [isCloud: false, endpoint: apiUrl, auth: null]
+    }
+
+    private boolean isCloudEndpoint(String apiUrl) {
+        return getCloudEndpointInfo(apiUrl).isCloud
+    }
+
+    protected Map callUserInfoApi(String accessToken, String apiUrl) {
+        final client = createHttpClient(accessToken)
+        final request = HttpRequest.newBuilder()
+            .uri(URI.create("${apiUrl}/user-info"))
+            .GET()
+            .build()
+
+        final response = client.send(request, HttpResponse.BodyHandlers.ofString())
+
+        if (response.statusCode() != 200) {
+            final error = response.body() ?: "HTTP ${response.statusCode()}"
+            throw new RuntimeException("Failed to get user info: ${error}")
+        }
+
+        final json = new JsonSlurper().parseText(response.body()) as Map
+        return json.user as Map
+    }
+
+    protected Path getConfigFile() {
+        return Const.APP_HOME_DIR.resolve('config')
+    }
+
+    protected Path getAuthFile() {
+        return Const.APP_HOME_DIR.resolve('seqera_auth.config')
```

Comment:
We tend to use kebab-case over snake_case, `seqera-auth.config` would be preferable